From 8bf4feec178ee4d1a6325cd651362f9e9877ced6 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Wed, 17 Feb 2016 11:25:12 +0200
Subject: [PATCH 0871/1240] tools: xdb: a3700: Add example XDB script

- Remove outdated A3700 XDB script
- Add example XDB script for A3700 board bringup
- Add static DDR configuration script for A3700 (600MHz)

Change-Id: I928bb75033ff12956cb11e08e8c43599520362fb
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27581
Reviewed-by: Haim Boot <hayim@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 .../debugger_scripts/XDB/QT_Hyperion_ARMv8_UP.xsf  |  16 ----
 tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb   | 101 +++++++++++++++++++++
 .../marvell/debugger_scripts/XDB/a3700_static.xdb  |  29 ++++++
 3 files changed, 130 insertions(+), 16 deletions(-)
 delete mode 100644 tools/marvell/debugger_scripts/XDB/QT_Hyperion_ARMv8_UP.xsf
 create mode 100644 tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb
 create mode 100644 tools/marvell/debugger_scripts/XDB/a3700_static.xdb

diff --git a/tools/marvell/debugger_scripts/XDB/QT_Hyperion_ARMv8_UP.xsf b/tools/marvell/debugger_scripts/XDB/QT_Hyperion_ARMv8_UP.xsf
deleted file mode 100644
index d87ba12..0000000
--- a/tools/marvell/debugger_scripts/XDB/QT_Hyperion_ARMv8_UP.xsf
+++ /dev/null
@@ -1,16 +0,0 @@
------------------------------------------------------------------------
- MARVELL INTERNATIONAL LTD., ON BEHALF OF ITSELF AND ITS WORLDWIDE
-AFFILIATES(COLLECTIVELY, "MARVELL"), MAKES NO WARRANTY OF ANY KIND WITH
-REGARD TO THIS MATERIAL, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
-OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
-MARVELL ASSUMES NO RESPONSIBILITY FOR ANY ERRORS THAT MAY
-APPEAR IN THIS DOCUMENT. MARVELL MAKES NO COMMITMENT TO
-UPDATE NOR TO KEEP CURRENT THE INFORMATION CONTAINED IN THIS DOCUMENT.
------------------------------------------------------------------------
-[PROJECT]
-Target=JTAG CS
-Template=true
-Version=2:2
-[SETTINGS]
-Global=
-CS=-L "tcp:127.0.0.1:3020:usb:" -target scanchain=FPGA_ARMV8_UP tckoverwrite=100kHz hotdebug device=mjc override=ap:1,debug:0x80030000,cti:0x80038000 option=0x20
diff --git a/tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb b/tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb
new file mode 100644
index 0000000..89abe78
--- /dev/null
+++ b/tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb
@@ -0,0 +1,101 @@
+! -----------------------------------------------------------------------
+!  MARVELL INTERNATIONAL LTD., ON BEHALF OF ITSELF AND ITS WORLDWIDE
+! AFFILIATES(COLLECTIVELY, "MARVELL"), MAKES NO WARRANTY OF ANY KIND WITH
+! REGARD TO THIS MATERIAL, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+! OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+! MARVELL ASSUMES NO RESPONSIBILITY FOR ANY ERRORS THAT MAY
+! APPEAR IN THIS DOCUMENT. MARVELL MAKES NO COMMITMENT TO
+! UPDATE NOR TO KEEP CURRENT THE INFORMATION CONTAINED IN THIS DOCUMENT.
+! -----------------------------------------------------------------------
+!
+! Running DDR initialization in 600MHz mode
+! All system clocks are set to default by the HW
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+SET VALUE /SIZE=LONG CORE:1(0xD0014008)=0x00404500
+SET VALUE /SIZE=LONG CORE:1(0xD0002000)=0x00010000
+SET VALUE /SIZE=LONG CORE:1(0xD0002004)=0x00000000
+SET VALUE /SIZE=LONG CORE:1(0xD0000340)=0x0f0f0fef
+SET VALUE /SIZE=LONG CORE:1(0xD0000344)=0x100000aa
+SET VALUE /SIZE=LONG CORE:1(0xD0000310)=0x200000
+SET VALUE /SIZE=LONG CORE:1(0xD0000304)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0000308)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0000200)=0xe0001
+SET VALUE /SIZE=LONG CORE:1(0xD0000204)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0000220)=0x13020532
+SET VALUE /SIZE=LONG CORE:1(0xD0000044)=0x30200
+SET VALUE /SIZE=LONG CORE:1(0xD00002c0)=0x6000
+SET VALUE /SIZE=LONG CORE:1(0xD00002c4)=0x00100020
+SET VALUE /SIZE=LONG CORE:1(0xD0000058)=0x143f
+SET VALUE /SIZE=LONG CORE:1(0xD0000048)=0x1
+SET VALUE /SIZE=LONG CORE:1(0xD0000180)=0x00010200
+SET VALUE /SIZE=LONG CORE:1(0xD0000050)=0x1ff
+SET VALUE /SIZE=LONG CORE:1(0xD000004c)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0000054)=0x480
+SET VALUE /SIZE=LONG CORE:1(0xD0000300)=0x708
+SET VALUE /SIZE=LONG CORE:1(0xD0000380)=0x1f5
+SET VALUE /SIZE=LONG CORE:1(0xD0000384)=0x3e9
+SET VALUE /SIZE=LONG CORE:1(0xD0000388)=0x9600043
+SET VALUE /SIZE=LONG CORE:1(0xD000038c)=0x200
+SET VALUE /SIZE=LONG CORE:1(0xD0000390)=0x400100
+SET VALUE /SIZE=LONG CORE:1(0xD0000394)=0x6b03cf
+SET VALUE /SIZE=LONG CORE:1(0xD0000398)=0x720200
+SET VALUE /SIZE=LONG CORE:1(0xD000039c)=0x120707
+SET VALUE /SIZE=LONG CORE:1(0xD00003a0)=0x40511
+SET VALUE /SIZE=LONG CORE:1(0xD00003a4)=0x1
+SET VALUE /SIZE=LONG CORE:1(0xD00003a8)=0xc04
+SET VALUE /SIZE=LONG CORE:1(0xD00003ac)=0x15210919
+SET VALUE /SIZE=LONG CORE:1(0xD00003b0)=0x90b0609
+SET VALUE /SIZE=LONG CORE:1(0xD00003b4)=0x4000600
+SET VALUE /SIZE=LONG CORE:1(0xD00003b8)=0x600
+SET VALUE /SIZE=LONG CORE:1(0xD00003bc)=0x2020404
+SET VALUE /SIZE=LONG CORE:1(0xD00003c0)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD00003c4)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD00003dc)=0x81239
+SET VALUE /SIZE=LONG CORE:1(0xD00002c8)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0000064)=0x0006
+SET VALUE /SIZE=LONG CORE:1(0xD0001004)=0x10077779
+SET VALUE /SIZE=LONG CORE:1(0xD0001008)=0x1ff00770
+SET VALUE /SIZE=LONG CORE:1(0xD000100c)=0x3f03fc77
+SET VALUE /SIZE=LONG CORE:1(0xD0001010)=0x100118
+SET VALUE /SIZE=LONG CORE:1(0xD0001028)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0001030)=0x3800000
+SET VALUE /SIZE=LONG CORE:1(0xD0001034)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0001040)=0x400
+SET VALUE /SIZE=LONG CORE:1(0xD00010c0)=0x80000001
+SET VALUE /SIZE=LONG CORE:1(0xD00010d0)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD00010e0)=0x11ff0
+SET VALUE /SIZE=LONG CORE:1(0xD0001090)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0001094)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0001098)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD000109c)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD00010a0)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD00010a4)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD00010a8)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD00010ac)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD00010b0)=0x0
+SET VALUE /SIZE=LONG CORE:1(0xD0001000)=0x44041
+SET VALUE /SIZE=LONG CORE:1(0xD0001014)=0x80200
+SET VALUE /SIZE=LONG CORE:1(0xD0001038)=0x2
+SET VALUE /SIZE=LONG CORE:1(0xD000103c)=0x10
+SET VALUE /SIZE=LONG CORE:1(0xD0001180)=0x20a
+SET VALUE /SIZE=LONG CORE:1(0xD0001184)=0x20a
+SET VALUE /SIZE=LONG CORE:1(0xD0001188)=0x20a
+SET VALUE /SIZE=LONG CORE:1(0xD000118c)=0x20a
+SET VALUE /SIZE=LONG CORE:1(0xD0001190)=0x20a
+SET VALUE /SIZE=LONG CORE:1(0xD0001194)=0x20a
+SET VALUE /SIZE=LONG CORE:1(0xD0001198)=0x20a
+SET VALUE /SIZE=LONG CORE:1(0xD000119c)=0x20a
+SET VALUE /SIZE=LONG CORE:1(0xD00011a0)=0x20a
+SET VALUE /SIZE=LONG CORE:1(0xD0001050)=0x8080000
+SET VALUE /SIZE=LONG CORE:1(0xD0001054)=0x8080000
+SET VALUE /SIZE=LONG CORE:1(0xD0001058)=0x8080000
+SET VALUE /SIZE=LONG CORE:1(0xD000105c)=0x8080000
+SET VALUE /SIZE=LONG CORE:1(0xD0001060)=0x8080000
+SET VALUE /SIZE=LONG CORE:1(0xD0001064)=0x8080000
+SET VALUE /SIZE=LONG CORE:1(0xD0001068)=0x8080000
+SET VALUE /SIZE=LONG CORE:1(0xD000106c)=0x8080000
+SET VALUE /SIZE=LONG CORE:1(0xD0001070)=0x8080000
+SET VALUE /SIZE=LONG CORE:1(0xD0001020)=0x20000000
+SET VALUE /SIZE=LONG CORE:1(0xD0001020)=0x40000000
+SET VALUE /SIZE=LONG CORE:1(0xD0001020)=0x80000000
+SET VALUE /SIZE=LONG CORE:1(0xD0000020)=0x11000001
diff --git a/tools/marvell/debugger_scripts/XDB/a3700_static.xdb b/tools/marvell/debugger_scripts/XDB/a3700_static.xdb
new file mode 100644
index 0000000..eb98c6b
--- /dev/null
+++ b/tools/marvell/debugger_scripts/XDB/a3700_static.xdb
@@ -0,0 +1,29 @@
+! -----------------------------------------------------------------------
+!  MARVELL INTERNATIONAL LTD., ON BEHALF OF ITSELF AND ITS WORLDWIDE
+! AFFILIATES(COLLECTIVELY, "MARVELL"), MAKES NO WARRANTY OF ANY KIND WITH
+! REGARD TO THIS MATERIAL, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+! OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+! MARVELL ASSUMES NO RESPONSIBILITY FOR ANY ERRORS THAT MAY
+! APPEAR IN THIS DOCUMENT. MARVELL MAKES NO COMMITMENT TO
+! UPDATE NOR TO KEEP CURRENT THE INFORMATION CONTAINED IN THIS DOCUMENT.
+! -----------------------------------------------------------------------
+! creation date: Tue Feb 16 17:11:27 2016
+
+! Activate memory access control in order to avoid XDB out of memory errors
+SET MAP /NAME="SPL" /READ="8,16,32,64,128" /WRITE="8,16,32,64,128" /SHAREDID=0 /PHYSICAL=ON /VIRTUAL=ON 0xF0000000 TO 0xF001FFFF
+SET MAP /NAME="DDR" /READ="8,16,32,64,128" /WRITE="8,16,32,64,128" /SHAREDID=1 /PHYSICAL=ON /VIRTUAL=ON 0x00000000 TO 0x1FFFFFFF
+SET MAP /NAME="REG" /READ="8,16,32,64,128" /WRITE="8,16,32,64,128" /SHAREDID=2 /PHYSICAL=ON /VIRTUAL=ON 0xD0000000 TO 0xDFFFFFFF
+SET MAP /NAME="ROM" /READ="8,16,32,64,128" /WRITE="8,16,32,64,128" /SHAREDID=3 /PHYSICAL=ON /VIRTUAL=ON 0xFFFF0000 TO 0xFFFFFFFF
+SET MACCESSCTRL 1
+
+! Run DDR initialization
+BATCH "./tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb"
+
+! Load SPL and u-boot
+LOAD /SEGMENT /DEBUG /GLOBAL /INIT="" OF "./spl/u-boot-spl"
+LOAD /BINARY /ADDRESS=0x0 OF "./u-boot.bin"
+LOAD /SEGMENT /DEBUG /GLOBAL /NOLOAD OF "./u-boot"
+
+! Start from SPL
+SET REGISTER PC=0xF0000000
+RUN
-- 
1.9.1

